### PR TITLE
Feat/devnet epoch 2.2

### PR DESCRIPTION
**Pull request by lgalabru**

### CODE REVIEW

**Review by obycode — DISMISSED** [review]
👍

**Review by obycode — APPROVED** [review]

File: components/clarinet-deployments/src/requirements.rs [hunk 1]
```diff
@@ -122,12 +122,18 @@ pub async fn retrieve_contract(
 
 pub const MAINNET_20_START_HEIGHT: u32 = 1;
 pub const MAINNET_2_05_START_HEIGHT: u32 = 40_607;
-// TODO: This is estimated. Replace with exact height once 2.1 is activated.
-pub const MAINNET_21_START_HEIGHT: u32 = 99_564;
-pub const TESTNET_20_START_HEIGHT: u32 = 1;
+pub const MAINNET_21_START_HEIGHT: u32 = 99_113;
+pub const MAINNET_22_START_HEIGHT: u32 = 103_900;
+pub const MAINNET_23_START_HEIGHT: u32 = 104_359;
+// TODO: This is estimated. Replace with exact height once 2.4 is activated.
```
Comment (hugocaillard) [inline components/clarinet-deployments/src/requirements.rs:128]:
```suggestion

```

File: components/stacks-network/src/chains_coordinator.rs [hunk 2]
```diff
@@ -105,23 +106,24 @@ impl DevnetEventObserverConfig {
...
+            bitcoind_rpc_password: devnet_config.bitcoin_node_password.clone(),
+            bitcoind_rpc_url: format!("http://{}", services_map_hosts.bitcoin_node_host),
+            bitcoin_block_signaling: BitcoinBlockSignaling::Stacks("http://0.0.0.0:20443".into()),
             stacks_node_rpc_url: format!("http://{}", services_map_hosts.stacks_node_host),
             operators: HashSet::new(),
             display_logs: true,
             cache_path: devnet_config.working_dir.to_string(),
             bitcoin_network: BitcoinNetwork::Regtest,
             stacks_network: chainhook_event_observer::chainhook_types::StacksNetwork::Devnet,
+            ordinals_enabled: false,
```
Comment (hugocaillard) [inline components/stacks-network/src/chains_coordinator.rs:126]: @lgalabru 
Could you please double-check here the default values I added? (including bitcoind_* and bitcoin_block_signaling)
Thx

File: components/stacks-rpc-client/src/rpc_client.rs [hunk 3]
```diff
@@ -67,7 +67,33 @@ pub struct PoxInfo {
...
+            reward_phase_block_length: 2000,
+            reward_slots: 4000,
+            total_liquid_supply_ustx: 1368787887756275,
+            ..Default::default()
+        }
+    }
+
+    pub fn testnet_default() -> PoxInfo {
+        PoxInfo {
+            contract_id: "ST000000000000000000002AMW42H.pox-2".into(),
```
Comment (hugocaillard) [inline components/stacks-rpc-client/src/rpc_client.rs:85]: Why not pox-3? cc @obycode @lgalabru

**Review by hugocaillard — COMMENTED** [review]
Left a couple of questions

File: components/stacks-rpc-client/src/rpc_client.rs (hunk 3 above)
Comment (obycode) [reply components/stacks-rpc-client/src/rpc_client.rs:85]: I don't think these defaults are actually used anywhere. I think they can just be deleted so we don't need to keep them in sync. I'll check and then add a commit if that's right.

Comment (hugocaillard) [reply components/stacks-rpc-client/src/rpc_client.rs:85]: The @obycode

**Review by obycode — APPROVED** [review]
LGTM! Thanks @hugocaillard. I just had one minor comment.

**Comment by obycode** [comment]
I just realized, we also need to publish the stacks-node image that works with epoch 2.4 and update the default setting.

File: components/stacks-rpc-client/src/rpc_client.rs (hunk 3 above)
Comment (lgalabru) [reply components/stacks-rpc-client/src/rpc_client.rs]: Maybe we could have @janniks cherrypicking @MicaiahReid's fix, rebase and force push in this branch?

Comment (obycode) [reply components/stacks-rpc-client/src/rpc_client.rs]: Oh, I missed your earlier comment. Yeah, feel free to back out or overwrite my commit that deleted this.

File: Cargo.toml [hunk 4]
```diff
@@ -14,3 +14,6 @@ members = [
     "components/stacks-rpc-client",
 ]
 default-members = ["components/clarinet-cli"]
+
+# [patch.crates-io]
+# "clarity-vm" = { path = "../stacks-blockchain/clarity" }
```
[the remark this answers is not in the document]
Comment (hugocaillard) [reply Cargo.toml:19]: Yeah! It took me some time to understand that it only works in the root `cargo.toml`. But this is definitely something we can generalize across the repos and components 👍

**Review by obycode — DISMISSED** [review]

**Review by hugocaillard — APPROVED** [review]

**Comment by hugocaillard** [comment]
Let's merge the branch, even though it breaks the CI, it's not directly related to the VM upgrade. A PR with the fix is coming.
I'll squash to keep the history clean
cc @lgalabru

**Pull request merged**